subsys/bluetooth/controller/hci/hci.c

```diff
@@ -103,6 +103,55 @@ static uint64_t event_mask = DEFAULT_EVENT_MASK;
 static uint64_t event_mask_page_2 = DEFAULT_EVENT_MASK_PAGE_2;
 static uint64_t le_event_mask = DEFAULT_LE_EVENT_MASK;
 
+static struct net_buf *cmd_complete_status(uint8_t status);
+
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+#if defined(CONFIG_BT_HCI_RAW)
+static int adv_intf_ensure_legacy(struct net_buf **cc_evt)
```

Comment:
Can we rename those functions to something a bit clearer
Perhaps

```
ll_adv_cmds_set()
adv_cmds_ensure_<>()
```

Reply:
> Can we rename those functions to something a bit clearer
> Perhaps
> 
> ```
> ll_adv_cmds_set()
> adv_cmds_check(bool legacy, ...)
> ```

I'm fine with either version and don't have any better name for that setting, so if @cvinayak also agrees I'll just change in next version

Reply:
@carlescufi can we agree to have two explicit functions? `adv_cmd_legacy_check(struct net_buf **cc_evt)` and `adv_cmd_ext_check(struct net_buf **cc_evt)`; this will avoid the additional parameter passing, unless all callers pass a constant value and are all in the same file. (my belief is link time optimizations is not present in zephyr?).

Reply:
@cvinayak this will increase code size, LTO is not enabled in Zephyr. I'd rather have a single function to be honest.

Reply:
@carlescufi no strong objections, I see @andrzej-kaczmarek has split into two separate functions, if he decides that there is not code size increase, I will go with his choice/decision.